### PR TITLE
clean_checkout input to handle cleaning after checkout

### PR DESCRIPTION
**Pull request by aliencaocao**
### `clean_checkout` input
- default: `true`
- If true, perform a clean checkout; if false, skip cleaning. Cleaning will remove all existing local files not in the repository during checkout. If you use utilities like pyinstaller-versionfile, set this to false.

This fixes issue of user wanting to use their own version text file but it is being cleaned by the checkout and very hard to debug.
See https://github.com/aliencaocao/netease_cloudmusic_discord_rpc/issues/4

### CODE REVIEW

**Comment by sayyid5416** [comment]
@aliencaocao Add this feature as **optional**. Add an input which controls if `clean` will be `True` or `False`

**Comment by aliencaocao** [comment]
@sayyid5416 is this ok?

**Comment by sayyid5416** [comment]
add in readme and other places as well, wherever necessary. I'll check later. Thanks for contribution btw.

**Comment by aliencaocao** [comment]
im not sure where else is needed except for readme. just updated it

**Comment by sayyid5416** [comment]
By default, `clean` should be `true` as it's `true` by default in `actions/checkout`

File: README.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ You can also use any major tags like `@v1` for any `@v1.*.*`
   | `pyinstaller_ver`     | -        | Specific pyinstaller version you want to use *(with proper signs, like `==5.13.2`)*
   | `exe_path`            | ./dist   | Path on runner-os, where executable will be stored
   | `upload_exe_with_name`| -        | Upload exe_ artifact with this name. Else, it won't upload
+  | `clean_checkout`      | false    | If true, perform a clean checkout; if false, skip cleaning. Cleaning will remove all existing local files not in the repository during checkout. If you use utilities like pyinstaller-versionfile, set this to false.
```
Comment (sayyid5416) [inline README.md]: by default: `true`

File: action.yml [hunk 2]
```diff
@@ -38,6 +38,9 @@ inputs:
   upload_exe_with_name:
     description: If passed, uploads executable artifact  with this name. Else, artifact won't be uploaded.
     default: ''
+  clean_checkout:
+    description: 'If true, perform a clean checkout; if false, skip cleaning. Cleaning will remove all existing local files not in the repository during checkout. If you use utilities like pyinstaller-versionfile, set this to false.'
+    default: false
```
Comment (sayyid5416) [inline action.yml]: by default: `true`

**Comment by aliencaocao** [comment]
done

**Review by sayyid5416 — APPROVED** [review]

**Comment by sayyid5416** [comment]
Do you need to add anything else to this PR?

**Comment by aliencaocao** [comment]
Nope

**Pull request merged**